api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/RoutineCalculation.java

```diff
@@ -125,110 +112,29 @@ public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Obj
 					ListResult vlObsResult = (ListResult) patientHavingVL.get(pId);
 					
 					List<Obs> viralLoadForPatientTakenWithin12Months = new ArrayList<Obs>();
-					List<Obs> vLoadList = new ArrayList<Obs>();
 					
-					if (vlObsResult != null && !vlObsResult.isEmpty()) {
-						vLoadList = EptsCalculationUtils.extractResultValues(vlObsResult);
-						
-						// populate viralLoadForPatientTakenWithin12Months with obs which fall within
-						// the 12month window
-						if (vLoadList.size() > 0) {
-							for (Obs obs : vLoadList) {
-								if (obs != null && obs.getObsDatetime().after(latestVlLowerDateLimit)
-								        && obs.getObsDatetime().before(context.getNow())) {
-									viralLoadForPatientTakenWithin12Months.add(obs);
-								}
-							}
-						}
-					}
+					List<Obs> vLoadList = getViralLoadForPatientTakenWithin12Months(context, latestVlLowerDateLimit,
+					    vlObsResult, viralLoadForPatientTakenWithin12Months);
 					
 					// find out for criteria 1 a
-					if (viralLoadForPatientTakenWithin12Months.size() >= 1) {
-						// the patients should be 6 to 9 months after ART initiation
-						// get the obs date for this VL and compare that with the provided dates
-						
-						for (Obs vlObs : viralLoadForPatientTakenWithin12Months) {
-							if (vlObs != null && vlObs.getObsDatetime() != null) {
-								Date vlDate = vlObs.getObsDatetime();
-								if (criteria.equals(PatientsOnRoutineEnum.ADULTCHILDREN)) {
-									if (EptsCalculationUtils.monthsSince(artInitiationDate, vlDate) > 6
-									        && EptsCalculationUtils.monthsSince(artInitiationDate, vlDate) <= 9) {
-										isOnRoutine = true;
-										break;
-									}
-								} else if (criteria.equals(PatientsOnRoutineEnum.BREASTFEEDINGPREGNANT)) {
-									if (EptsCalculationUtils.monthsSince(vlDate, artInitiationDate) > 3
-									        && EptsCalculationUtils.monthsSince(vlDate, artInitiationDate) <= 6) {
-										isOnRoutine = true;
-										break;
-									}
-								}
-							}
-						}
+					// the patients should be 6 to 9 months after ART initiation
+					// get the obs date for this VL and compare that with the provided dates
+					if (viralLoadForPatientTakenWithin12Months.size() >= 1
```

Comment:
I don't think we need this verification **viralLoadForPatientTakenWithin12Months >=1** once we iterate accross this collection in method **isOnRoutineCriteria1()** to determinate the value of **isOnRoutine**

Reply:
@Guimino done